spec/rubocop/cop/sorbet/obsolete_strict_memoization_spec.rb

```diff
@@ -221,4 +221,25 @@ def foo
       end
     end
   end
+
+  describe "a mistaken variant of the obsolete memoization pattern" do
+    it "registers an offence and autocorrects" do
+      # This variant would have been a mistake, which woudl have caused the memoized value to be discarded and recomputed
+      # on every call. We can fix it up into the working version.
+
+      expect_offense(<<~RUBY)
+        def foo
+          @foo = T.let(nil, T.nilable(Foo))
+          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ This two-stage workaround for memoization in `#typed: strict` files is no longer necessary. See https://sorbet.org/docs/type-assertions#put-type-assertions-behind-memoization.
```

Comment:
Should we change the error message? This isn't just changing the syntax, it's fixing a bug.